pkg/build/builder/docker.go

```diff
@@ -69,6 +72,20 @@ func (d *DockerBuilder) Build() error {
 	tag := d.build.Parameters.Output.DockerImageReference
 	defer removeImage(d.dockerClient, tag)
 	if len(d.build.Parameters.Output.DockerImageReference) != 0 {
+		// TODO: This should use ImageRepository ObjectReference in future
```

Comment:
I don't think you need this... just like we resolve From to an ImageReference before creating the Build object, we should always resolve To to an ImageReference before creating the Build object.  That will ensure you can rerun the build w/ the same inputs+outputs even if the ImageRepo referred to by To or From has changed.

it should be invalid to specify To or From in the Build object itself (unfortunately our validation logic is shared right now so we need to refactor to make that happen.  See my comments here: https://github.com/openshift/origin/pull/1406#issuecomment-84725616